JS-week-2/candy_helper.js

```diff
@@ -0,0 +1,43 @@
+let boughtCandy = [];
+
+function addCandy(candyType, weight){
+    let cadies = ['sweet','chocolate','toffee','chewing_gum'];
+    let candyPricePerGram = [0.5, 0.7, 0.3, 0.03];
+
+   // let candyIndex = cadies.findIndex((element) => element === candyType);
```

Comment:
Using `findIndex` is very clever! I assume that you have commented it out to deal with cases when the candy type provided isn't found in the array. The documentation for this method says:

> The index of the first element in the array that passes the test. Otherwise, -1.

So, I can suggest adding an extra condition block to check if `candyIndex` is not `-1`, and if it is then simply return from the function. 

P.S. The solution you have provided below also works, so it's completely up to you which one to use.